src/h5file.cpp

```diff
@@ -803,8 +813,9 @@ static void _read_chunk(hid_t data_id, int rank, const size_t *chunk_start,
 }
 
 void h5file::read_chunk(int rank, const size_t *chunk_start, const size_t *chunk_dims,
-                        realnum *data) {
-  _read_chunk(HID(cur_id), rank, chunk_start, chunk_dims, REALNUM_H5T, (void *)data);
+                        void *data, bool single_precision) {
+  _read_chunk(HID(cur_id), rank, chunk_start, chunk_dims,
+              single_precision ? H5T_NATIVE_FLOAT : H5T_NATIVE_DOUBLE, data);
 }
```

Comment:
It would be clearer to just have two methods here, one for `float *` and one for `double *`.  That way the compiler will choose the correct method depending on what argument is passed.

Similarly elsewhere.

Reply:
Done.